UI Development/chartPage.pde

```diff
@@ -1,12 +1,204 @@
+      }
+      stroke(255);
+      strokeWeight(3);
+      arc(width/2, height/2, diameter, diameter, lastAngle, lastAngle+radians(angles[i]), PIE);
+      lastAngle += radians(angles[i]);
+      //println(lastAngle);
+    }
+    textAlign(CENTER);
+    textSize(40);
+    fill(WHITE_SWATCH);
+    text("Percent Delayed for " + gui.inputFilters.get(0), Gui.SCREEN_WIDTH/2, (Gui.SCREEN_HEIGHT*indentation));
+    
+    textSize(10);
+    float labelX = width/2 + cos(0) * (diameter/2 + 60); 
+    float labelY = height/2 + sin(0) * (diameter/2 + 20);
+    String percentageText = String.format("%.2f", (angles[0] / 360) * 100); // Format percentage to one decimal place
+    text(percentageText + "% were delayed", labelX, labelY);
+    
+    if(angles[0] < angles[1])
+    {
+      labelX = width/2 + cos(radians(180)) * (diameter/2 + 60); 
+      labelY = height/2 + sin(radians(180)) * (diameter/2 + 20);
+      println("here");
+    }
+    else
+    {
+      labelX = width/2 + cos(radians(angles[0])) * (diameter/2 + 60); 
+      labelY = height/2 + sin(radians(angles[0])) * (diameter/2 + 20);
+    }
+    percentageText = String.format("%.2f", (angles[1] / 360) * 100);
+    text(percentageText + "% were not\ndelayed", labelX, labelY);
+}
+
+
+
+  void pieChartForDates(int[] frequency, String[] labels)
+  {
+    colorMode(HSB, 360, 50, 10); // Set color mode to HSB
+    float lastAngle = 0;
+    int runningTotal = 0;
+    float[] angles = new float[frequency.length];
+    
+    for (int index = 0; index < frequency.length; index++)
+    {
+      runningTotal += frequency[index];
+    }
+    println(runningTotal);
+    for (int index = 0; index < angles.length; index++)
+    {
+      float tempValue = ((float)(frequency[index]) / runningTotal) * 360;
+     // float tempAngle = (float)tempValue;
+      angles[index] = tempValue;
+      println(tempValue);
+    }
+    for (int i = 0; i < frequency.length; i++) 
+    {
+      float hue = (lastAngle / TWO_PI) * 360; // Calculate hue value based on the angle      fill(hue, 30, 90); // Set fill color based on hue
+      fill(hue, 30, 90); // Set fill color based on hue with lower saturation and higher brightness
+      stroke(0);
+      strokeWeight(3);
+      arc(width/2, height/2, diameter, diameter, lastAngle, lastAngle+radians((float)angles[i]), PIE);
+      float labelX = width/2 + cos(lastAngle) * (diameter/2 + 20); 
+      float labelY = height/2 + sin(lastAngle) * (diameter/2 + 20);
+      textSize(20);
+      fill(255);
+      textAlign(CENTER, CENTER);
+      text(labels[i], labelX, labelY);
+      lastAngle += radians((float)angles[i]);
+    }
+    textAlign(CENTER);
+    textSize(40);
+    fill(WHITE_SWATCH);
+    text("FLIGHT FREQUENCIES:\n" + "JANUARY", Gui.SCREEN_WIDTH/2, (Gui.SCREEN_HEIGHT*indentation));
+  }
 
-  public void setup() {}
+  public void setup() {
+    background(BG_COLOR);
+    super.setHeaderFont("PPSupplyMono-Regular.otf", 20);
+  }
 
   public void draw() {
```

Comment:
Where all the back-end class objects (Airports, Dates, Delays) should be drawn. Insert the if-block I commented on in this function.